src/views/ViewResource.js

```diff
@@ -94,12 +100,24 @@ function FileTable(props) {
 }
 
 export default function ViewResource() {
+  let [resource, setResource] = useState(emptyRes);
```

Comment:
can we make this default to null (rather than empty res) and then just render a spinner if it hasn't loaded yet